interfaces/astarte_interface.go

```diff
@@ -321,6 +321,54 @@ type AstarteInterface struct {
 	Mappings          []AstarteInterfaceMapping   `json:"mappings"`
 }
 
+// requiredAstarteInterface is an helper struct used for validating required fields when unmarshalling an
+// astarte interface. Its fields are defined as pointers so that it is possible determining if any field is
+// present and valid.
+type requiredAstarteInterface struct {
+	Name         *string                           `json:"interface_name"`
+	MajorVersion *int                              `json:"version_major"`
+	MinorVersion *int                              `json:"version_minor"`
+	Mappings     []requiredAstarteInterfaceMapping `json:"mappings"`
+}
```

Comment:
Type and ownership should be required, too (see [the docs](https://docs.astarte-platform.org/latest/040-interface_schema.html#interface)).